grails-app/services/org/broadinstitute/orsp/StorageProviderService.groovy

```diff
@@ -173,14 +173,15 @@ class StorageProviderService implements Status {
                 creator: displayName,
                 username: userName,
                 creationDate: new Date(),
-                status: DocumentStatus.PENDING.status
+                status: DocumentStatus.PENDING.status,
+                consentCollectionLinkId: consentCollectionLink?.id
         )
         if (saveStorageDocument(document, file.getInputStream())) {
             persistenceService.saveEvent(
-                    document.projectKey,
+                    document.projectKey == null ? consentCollectionLink.consentKey : document.projectKey,
```

Comment:
This poses an interesting question - and I agree with this solution - but the question is, what is the CCL document more closely associated with? I think this is the right answer.